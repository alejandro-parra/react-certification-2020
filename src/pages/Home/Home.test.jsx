import React from 'react';
// import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HomePage from './Home.page';

// import mockVideos from '../../data/mock-videos.json';

describe('HomePage UI', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomePage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('changes screen when clicking on a snippet', async () => {
  //   render(<HomePage />);
  //   await waitFor(() => {
  //     screen.getByRole('button');
  //   });
  //   act(() => {
  //     fireEvent.click(screen.getByRole('button')); // cards have roles of button
  //   });
  //   const role = 'youtube-video';
  //   expect(screen.getByRole(role)).toBeTruthy();
  // });
});
