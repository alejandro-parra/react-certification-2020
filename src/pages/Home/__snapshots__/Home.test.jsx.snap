// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage UI renders correctly 1`] = `
Array [
  <div
    className="sc-kstrdz VHRkH"
  >
    <div
      className="sc-fodVxV sc-fFubgz bPTrMk fhliTH"
    >
      <img
        alt="Logo"
        className="sc-hBEYos kzrmER"
        onClick={[Function]}
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <div
      className="sc-jrAGrp idCptY"
    >
      <input
        className="sc-kEjbxe keRHpK"
        onChange={[Function]}
        placeholder="Busca un video"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root sc-iqHYGH QaGG"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      className="sc-fodVxV sc-bkzZxe bPTrMk OsNsC"
    >
      <div
        className="MuiFormGroup-root"
      >
        <label
          className="MuiFormControlLabel-root"
          role="switch"
        >
          <span
            className="MuiSwitch-root"
          >
            <span
              aria-disabled={false}
              className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorSecondary"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label"
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input-4 MuiSwitch-input"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="MuiSwitch-thumb"
                />
              </span>
            </span>
            <span
              className="MuiSwitch-track"
            />
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Claro
          </span>
        </label>
      </div>
      <div
        className="sc-crrsfI klUkej"
      >
        <div
          className="sc-dQppl cmeZYW"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <div
    className="sc-pFZIQ giDqbV"
  />,
]
`;
